### PR TITLE
Add error message when trying to use `wp rewrite flush --hard` on a multisite install

### PR DESCRIPTION
**Pull request by rodrigoprimo**
WordPress don't generate .htaccess for multisite installs:

https://github.com/WordPress/WordPress/blob/ce557062f4123d8513378cf415b4e8b612c33ccc/wp-admin/includes/misc.php#L162

### CODE REVIEW

File: features/rewrite.feature [hunk 1]
```diff
@@ -62,3 +62,16 @@ Feature: Manage WordPress rewrites
...
+    Given a WP multisite install
+    And a wp-cli.yml file:
+      """
+      apache_modules: [mod_rewrite]
+      """
+
+    When I try `wp rewrite flush --hard`
+    Then STDERR should be:
+      """
+      Error: WordPress can't generate .htaccess file for a multisite install.
```
Comment (danielbachhuber) [inline features/rewrite.feature]: Can we throw a warning, instead of a hard error, so we can have the option refreshed still?

**Comment by rodrigoprimo** [comment]
Unfortunately this change won't affect `wp rewrite structure --hard` since this commands calls `wp rewrite flush` via `WP_CLI::launch_self()` without checking for errors.

**Comment by rodrigoprimo** [comment]
@danielbachhuber I have replaced the error with a warning as you suggested.

As I mentioned above ideally this warning message should appear when using `wp rewrite structure --hard` as well. I tried to make it work by changing this line https://github.com/wp-cli/wp-cli/blob/master/php/commands/rewrite.php#L125 to:

```
$process = \WP_CLI::launch_self( 'rewrite flush', array(), $new_assoc_args, false, true );
```

But then there seems to be a problem with `WP_CLI::launch_self` as it is returning an unexpected warning message saying that mod_rewrite is not in apache_modules configuration. I think that this method is not properly loading WP-CLI config when launching a new WP-CLI process but I haven't investigated this further.

```
class WP_CLI\ProcessRun#140 (6) {
  public $stdout =>
  string(0) ""
  public $stderr =>
  string(161) "Warning: Regenerating a .htaccess file requires special configuration. See usage docs.
Warning: WordPress can't generate .htaccess file for a multisite install.
"
  public $return_code =>
  int(0)
  public $command =>
  string(138) "/usr/bin/php5 /home/rodrigo/devel/wp-cli/bin/../php/boot-fs.php rewrite flush   --hard --path='/home/rodrigo/devel/wordpress-develop/src/'"
  public $cwd =>
  NULL
  public $env =>
  array(0) {
  }
}
```

Thoughts?

**Comment by rodrigoprimo** [comment]
On a related note, do you happen to know why core doesn't generate .htaccess for multisite?

**Comment by danielbachhuber** [comment]
For subdomain multisite on Apache, a new VirtualHost record will need to be created. Not sure about subdirectories.

**Comment by danielbachhuber** [comment]
> I think that this method is not properly loading WP-CLI config when launching a new WP-CLI process but I haven't investigated this further.

It doesn't, it just persists specific runtime args: https://github.com/wp-cli/wp-cli/blob/master/php/class-wp-cli.php#L405

I think the most appropriate thing to do would be to pass `apache_mods` as another runtime arg to use, specifically for this command.

We can go ahead and land this as it is.

**Pull request merged**